cips/cip-040.md

```diff
@@ -0,0 +1,142 @@
+- Simplify KMS implementations by avoiding message content validation
+
+**Chain ID Inclusion**: Including the chain_id prevents signature reuse across different networks, maintaining security boundaries between chains.
+
+**Unique ID Field**: The optional unique_id field allows protocols to differentiate between different message types or contexts without requiring separate interface extensions.
+
+**No Double-Signing Protection**: Unlike consensus messages, P2P messages do not require double-signing protection as they do not affect chain safety. This simplification reduces implementation complexity across KMS providers.
+
+### Alternative Approaches Considered
+
+**Message Type Enumeration**: An alternative approach would enumerate specific message types (FMO, Vacuum!, etc.). However, the hash-based approach provides greater flexibility for future protocol extensions without requiring interface changes.
+
+**Separate Interface**: Creating a completely separate interface was considered but rejected to maintain consistency with existing KMS integrations and avoid fragmenting the signing infrastructure.
+
+## Backwards Compatibility
+
+This proposal is fully backwards compatible. Existing KMS implementations will continue to function normally, as the new message types use previously unused field numbers in the protobuf oneof union. Legacy implementations that do not support the new message types will return appropriate error responses, allowing calling code to handle the limitation gracefully.
+
+## Security Considerations
+
+### Key Reuse
+
+This proposal reuses validator consensus keys for signing offchain messages. While this introduces additional key usage, the risk is mitigated by:
+
+1. **Hash-based signing**: Only pre-computed hashes are signed, preventing manipulation of the signing process
+2. **Chain ID scoping**: Signatures are bound to specific chain contexts
+3. **No state modification**: P2P message signatures do not affect blockchain state
+
+### Signature Verification
+
+Applications using this interface MUST implement proper signature verification including:
+
+- Validating the signer's public key against the validator set
+- Verifying the chain_id matches the expected network
+- Confirming the hash corresponds to the expected message content
+
+### DoS Protection
+
+KMS implementations SHOULD implement rate limiting for P2P signing requests to prevent resource exhaustion attacks against the signing infrastructure.
+
+### Key Compromise
+
+In the event of key compromise, the impact extends beyond consensus to any protocols using this signing interface. Monitoring systems SHOULD track unusual signing patterns across both consensus and P2P message types.
+
+## Reference Implementation
+
+A reference implementation is available in CometBFT PR [#5138](https://github.com/cometbft/cometbft/pull/5138). This implementation has been tested on multiple private testnets and the Mamo testnet, demonstrating practical viability.
```

Comment:
This PR appears to allow signing over arbitrary bytes not just hashes so it seems slightly divergent from this spec.

Also the requests are named differently:
- sign_raw_bytes_request in the CometBFT PR
-  sign_p2p_message_request in this spec

Reply:
Yes, we should use the raw messages because ed25519 is vulnerable for hashes